### PR TITLE
[DOCS] Added contributing section to the documentation

### PR DESCRIPTION
**Pull request by MagdaDziadosz**
Added contributing section to the documentation:

- [x] Overview 
    - [x] Tracking issues - included in overview
-  [x] Workflow
    - [x] Fork the project
    - [x] Commit and push the change
    - [x] Open a pull request
- [x] Coding guidelines - waiting for dev input
- [x] Testing
    - [x] Contributing to documentation - included in workflow and testing

Jira ticket: [HEL-27](https://mirantis.jira.com/browse/HEL-27)

Add the Contributing section

### CODE REVIEW

File: docs/contributing/contribute-testing.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+   In the repository root directory, make sure that:
+
+    * `make build && git diff --exit-code` runs successfully.  
+      Verifies that the build is working and that the generated source code
+      matches the one that's checked into source control.
+    * `make test` runs successfully.  
+      Verifies that all the unit tests pass.
+
+   Note that the last test may produce a false failing result, so it might fail on
```
Comment (makhov) [inline docs/contributing/contribute-testing.md:49]: This paragraph is redundant for k0smotron

File: docs/contributing/contribute-testing.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+## Open pull request
+
+### Draft mode
+
+You may open pull request in [a draft mode](https://github.blog/2019-02-14-introducing-draft-pull-requests).
+It will go through the automated testing, but the pull request will not be assigned for a review.
+Once a pull request is ready for review, transition it from the draft mode to notify the k0smotron team.
+
+### Conformance testing
```
Comment (makhov) [inline docs/contributing/contribute-testing.md:59]: This section is also redundant

File: README.md [hunk 3]
```diff
@@ -84,7 +84,7 @@ Explore the comprehensive [documentation](https://docs.k0smotron.io/stable) for
 
 ## Contributing
 
-Contributions to k0smotron are welcome! Read the [Contribution Guidelines](contributors/) to learn how you can participate in enhancing this tool.
+Contributions to k0smotron are welcome! Read the [Contribution Guidelines](docs/contributing/contribute-overview.md) to learn how you can participate in enhancing this tool.
```
Comment (KoryKessel-Mirantis) [inline README.md:87]: I suggest flipping the clauses in the second sentence to read:

To learn how you can contribute to the growth and enhancement of k0smotron, please read our [Contribution Guidelines](docs/contributing/contribute-overview.md).

**Review by makhov — APPROVED** [review]

File: docs/contributing/contribute-testing.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+1. The pull request has passed all the automated tests (style, build and conformance tests).
+2. Pull request commits have been signed with the `--signoff` option.
+3. Pull request was reviewed and approved by a code owner.
+4. Pull request is rebased against upstream main branch.
+
+## Cleanup local workspace
+
+To clean up the local workspace, run `make clean`.
+It cleans up all the intermediate files and directories created during the k0smotron build.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-testing.md:71]: Suggest:  To clean up the local workspace, run `make clean'. This removes all the intermediate files and directories that were created during the k0smotron build.

File: docs/contributing/contribute-workflow.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
+# k0smotron GitHub workflow
+
+You can contribute to [k0smotron GitHub repository](http://github.com/k0sproject/k0smotron)
+by creating pull requests.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:4]: Create pull requests to...

File: docs/contributing/contribute-workflow.md [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+
+## Fork the project
+
+1. Go to [k0smotron GitHub repository](http://github.com/k0sproject/k0smotron).
+2. On the top, right-hand side, click "Fork" and select your username for the fork destination.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:10]: Suggest "In the top right-hand corner, "

File: docs/contributing/contribute-workflow.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    ```shell
+    origin  https://github.com/k0sproject/k0smotron (fetch)
+    origin  no_push (push)
+    my_fork git@github.com:{ github_username }/k0smotron.git (fetch)
+    my_fork git@github.com:{ github_username }/k0smotron.git (push)
+    ```
+
+## Create and rebase feature branch
+
+1. Create a feature branch and switch to it:
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:49]: Suggest: "Create and switch to a..."

File: docs/contributing/contribute-workflow.md [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+Description that briefly explains the problem your commit is solving. 
+Focus on why you are making this change as opposed to how.
+Are there any consequences of this change? Here you can include them.
+
+Fixes: https://github.com/k0sproject/k0smotron/issues/373
+
+Signed-off-by: Name Lastname <user@example.com>
+```
+
+If you need to add some additional changes or tests use `commit --amend` command.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:98]: Suggest: "To add..."

File: docs/contributing/contribute-workflow.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+
+When everything is ready, push your changes to your fork's repository:
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:100]: Suggest: Push your changes to the fork repository.

File: docs/contributing/contribute-workflow.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+```shell
+git push --set-upstream my_fork my_feature_branch
+```
+
+## Open pull request
+
+Follow GitHub's documentation on how to [create a pull request from a fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:108]: Suggest: "Refer to the official GitHub documentation [Creating a pull request from a fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).

File: docs/contributing/contribute-workflow.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+### Wait for code review
+
+The k0smotron team will review your pull request.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:112]: Suggest: after 'pull request' insert 'and leave comments.'

File: docs/contributing/contribute-workflow.md [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+Commit changes made in response to review comments should be added to the same branch on your fork.
+
+Keep the PRs small if possible to speed up the review process.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:115]: Need commas.

File: docs/contributing/contribute-workflow.md [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+### Squash small commits
+
+Commits on your branch should represent meaningful milestones or units of work.
+Squash together small commits that contain typo fixes, rebases, review feedbacks,
+and so on.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:121]: Separate 'Squash' sentence out to a new paragraph and attach the sentence following to it.

File: docs/contributing/contribute-workflow.md [hunk 14]
```diff
@@ -0,0 +1,133 @@
...
+
+### Push final changes
+
+Once done, you can push the final commits to your branch:
+
+```shell
+git push --force
+```
+
+You can run multiple iteration of `rebase`/`push -f`, if needed.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-workflow.md:133]: If necessary, for whatever reason...

**Review by KoryKessel-Mirantis — CHANGES_REQUESTED** [review]

**Comment by juanluisvaladas** [comment]
LGTM both as is or with the suggestions work for me as well.
Needs rebase, feel free to ping me on slack for stamping it.

File: docs/contributing/contribute-overview.md [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+Thank you for taking the time to make a contribution.
+We are looking forward to working with you to make k0smotron better.
+
+The following document is a set of helpful guidelines and instructions for the contributors.
+
+## Report issue
+
+Before proceeding with the contribution, consider first discussing your change
+by [opening a GitHub issue in a k0smotron project](https://github.com/k0sproject/k0smotron/issues).
```
[the remark this answers is not in the document]
Comment (KoryKessel-Mirantis) [reply docs/contributing/contribute-overview.md]: LGTM

File: docs/contributing/contribute-overview.md [hunk 16]
```diff
@@ -0,0 +1,53 @@
+# Contributing to k0smotron
```
[the remark this answers is not in the document]
Comment (KoryKessel-Mirantis) [reply docs/contributing/contribute-overview.md]: LGTM

File: docs/contributing/contribute-overview.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+
+Thank you for taking the time to make a contribution.
+We are looking forward to working with you to make k0smotron better.
```
[the remark this answers is not in the document]
Comment (KoryKessel-Mirantis) [reply docs/contributing/contribute-overview.md]: LGTM

File: docs/contributing/contribute-overview.md [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+
+The following document is a set of helpful guidelines and instructions for the contributors.
```
[the remark this answers is not in the document]
Comment (KoryKessel-Mirantis) [reply docs/contributing/contribute-overview.md]: LGTM

File: docs/contributing/contribute-overview.md [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+
+## Report issue
+
+Before proceeding with the contribution, consider first discussing your change
+by [opening a GitHub issue in a k0smotron project](https://github.com/k0sproject/k0smotron/issues).
+
+To stay updated, you can subscribe to an issue to receive notifications about the latest comments.
```
[the remark this answers is not in the document]
Comment (KoryKessel-Mirantis) [reply docs/contributing/contribute-overview.md]: LGTM

File: docs/contributing/contribute-overview.md [hunk 20]
```diff
@@ -0,0 +1,53 @@
...
+## Test code
+
+All submitted PRs go through a set of tests and reviews. 
+Remember to run tests **before** submitting a PR.
+It saves your time on many possible review iterations and automated tests.
+The testing guidelines can be found in [testing guidelines](contribute-testing.md).
+
+## License
+
+By contributing, you agree that your contributions will be licensed as followed:
```
[the remark this answers is not in the document]
Comment (KoryKessel-Mirantis) [reply docs/contributing/contribute-overview.md]: 'followed' should be 'follows'

File: docs/contributing/contribute-testing.md [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+Once you decide the PR is ready for review, transition it out of draft mode to notify the k0smotron team.
+
+### Pre-requisites for merge
+
+In order for a pull request to be merged, the following conditions should exist:
+
+1. Pull request passes all the automated tests (style, build and conformance tests).
+2. Pull request commits are signed with the `--signoff` option.
+3. Pull request is reviewed and approved by a code owner.
+4. Pull request is rebased against upstream main branch.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-testing.md:61]: I suggest rethinking this edit.

First, can a PR be merged if any of these conditions are not fulfilled? The 'should' raises questions where perhaps questions should not be raised.

Second, if it is not a 'should' but is indeed a 'must', then the conditions should be presented in the past tense, as they were prior to this edit.

If the conditions are a 'must', I suggest:

For a pull request to be merged, all of the following must be true:

1. The PR has passed all of the automated tests (style, build and conformance tests).
2. All of the PR commits have been signed with the `--signoff` option.
3. The PR has been reviewed and approved by a code owner.
4. The PR has been rebased against upstream main branch.

File: docs/contributing/contribute-testing.md [hunk 22]
```diff
@@ -0,0 +1,70 @@
...
+
+## Cleanup local workspace
+
+To clean up the local workspace, run `make clean`.
+This removes all the intermediate files and directories created during the k0smotron build.
```
Comment (KoryKessel-Mirantis) [inline docs/contributing/contribute-testing.md:66]: Suggest:

Run 'make clean' to remove all the intermediate files and directories created locally during the k0smotron build.

**Review by KoryKessel-Mirantis — CHANGES_REQUESTED** [review]

**Comment by MagdaDziadosz** [comment]
@juanluisvaladas ready for final check

**Review by KoryKessel-Mirantis — APPROVED** [review]

**Pull request merged**